decide.go

```diff
@@ -68,6 +77,37 @@ func addNode(node *Node, expressions []string, object string) (*Node, error) {
 	return node, err
 }
 
+func nextExpression(expressions []string, match exp.Expression) (exp.Expression, []string, error) {
+	if len(expressions) == 0 {
+		return nil, nil, nil
+	}
+
+	if match == nil {
+		// No match, use first
+		expression, err := NewExpression(expressions[0])
+		if err != nil {
+			return nil, nil, err
+		}
+		return expression, expressions[1:], nil
+	}
+
+	// See if the expression exists in the list
+	e := make([]string, 0, len(expressions))
+	for i, exprstr := range expressions {
```

Comment:
At a later time, we may want to pass an index back instead of having to parse all the expressions for a match repeatedly